articles/user-guide/host-programs.md

```diff
@@ -0,0 +1,455 @@
+However, we can instruct the operation to be run on a specific target machine with the option `--simulator` (or the shorthand `-s`).
+
+For example, we could run it on [`ResourcesEstimator`](xref:microsoft.quantum.machines.resources-estimator):
+
+```bash
+dotnet run -n 4 -s ResourcesEstimator
+```
+
+The printed output is then
+
+```bash
+Metric          Sum
+CNOT            0
+QubitClifford   4
+R               0
+Measure         8
+T               0
+Depth           0
+Width           4
+BorrowedWidth   0
+```
+
+Note that the [`Reset`](xref:microsoft.quantum.intrinsic.reset) and [`ResetAll`](xref:microsoft.quantum.intrinsic.resetall) operations utilize measurement to reset the qubits to $\ket{0}$, hence the reported 8 measurements as opposed to the 4 which yield the actual results.
+Details on what these metrics indicate can be found [here](xref:microsoft.quantum.machines.resources-estimator#metrics-reported).
+
+### Command line execution summary
+<br/>
+<img src="./images/command_line_diagram.png" alt="Q# program from command line" width="700">
+
+
+## Q# with host programs
+
+With our Q# file in hand, an alternative to calling an operation directly from the command line is to use a *host program* in another classical language. 
+Specifically, this can be done with either Python or a .NET language such as C# or F# (for the sake of brevity we will only detail C# here).
+A little more setup is required to enable the interoperability, but those details can be found in the [install guides](xref:microsoft.quantum.install).
+
+In a nutshell, the situation now includes a host program file (e.g. `*.py` or `*.cs`) in the same location as our Q# file.
+It's now the *host* program that gets run, and in the course of its execution it calls specific Q# operations from the Q# file.
+The core of the interoperability is based on the Q# compiler making the contents of the Q# file accessible to the host program so that they can be called.
+
+One of the main benefits of using a host program is that the classical data returned by the Q# program can then be further processed in the host language.
+This could consist of some advanced data processing (e.g. something that can't be performed internally in Q#), and then calling further Q# actions based on those results, or something as simple as plotting the Q# results.
+
+The general scheme is shown here, and we discuss the specific implementations for Python and C# below. 
+A sample using an F# host program can be found at the [.NET interoperability samples](https://github.com/microsoft/Quantum/tree/master/samples/interoperability/dotnet).
+
+<br/>
+<img src="./images/host_program_diagram.png" alt="Q# program from a host program" width="700">
+
+> [!NOTE]
+> The `@EntryPoint()` attribute used for Q# command line applications cannot be used with host programs.
+> An error will be raised if it is present in the Q# file being called by a host. 
```

Comment:
@bettinaheim @samarsha would you mind commenting on this? Thank you.

Reply:
I think that the recommended way to use a custom host driver is to still use the `@EntryPoint()` attribute, but also create a separate project and add the Q# project as a reference. If you have a C# driver in the same project as a Q# program with an entry point, then the C# compiler sees two `Main` methods and doesn't know which one to choose.

For now, I think that not using the `@EntryPoint()` attribute and putting the C# driver in the same project would also still work, but I think mixing C# and Q# is something we want to avoid going forward.